src/AddressRegistry/StreetName/StreetNameAddress.cs

```diff
@@ -378,6 +379,44 @@ private void CorrectPostalCodeBecauseParentPostalCodeWasChanged(PostalCode posta
             Apply(new AddressPostalCodeWasCorrectedV2(_streetNamePersistentLocalId, AddressPersistentLocalId, postalCode));
         }
 
+        public void CorrectHouseNumber(HouseNumber houseNumber)
+        {
+            GuardNotRemovedAddress();
+
+            var validStatuses = new[] { AddressStatus.Proposed, AddressStatus.Current };
+
+            if (!validStatuses.Contains(Status))
+            {
+                throw new AddressHasInvalidStatusException();
+            }
+
+            if (BoxNumber is not null)
+            {
+                throw new HouseNumberToCorrectHasBoxNumberException();
+            }
+
+            var formatRegex = new Regex("^[1-9]([0-9]{0,8}([A-H]|[K-N]|[P]|[R-T]|[V-Z]){0,1}|[0-9]{0,9})$", RegexOptions.Compiled);
```

Comment:
move the regex somewhere central (static)?
Also should we also do this check in propose?

Reply:
Done